docu/Concepts/Snippets/1-Blockchain/Connector/Dezentralität.md

```diff
@@ -0,0 +1,70 @@
+
+Diese Transaktionsblöcke werden dann in Form einer Blockchain gespeichert. Sie werden durch einen running hash oder transactions hash 
+miteinander verbunden. Dieser wird aus dem vorherigen Hash, der Transaktionsnr, Empfangsdatum und der Signaturen berechnet. 
+Jeder der einen Node-Server betreibt hat dadurch eine einfache Möglichkeit zu überprüfen ob andere Node-Server einwandfrei funktionieren,
+indem er einfach den running hash von einer Transaktion auf seinem Node mit dem running hash derselben Transaktion (gleiche nr) des entsprechenden 
+Node-Servers vergleicht.
+
+### Wo werden nun die Daten genau gespeichert?
+Bei Bitcoin und Co. werden einfach alle Transaktionen bei allen Node Betreibern gespeichert. Das stößt aber schnell an physikalische Grenzen. 
+Daher werden die Daten bei Gradido in spearate Blockchains pro Community gespeichert. Dabei ist der Node-Server so ausgelegt, das er beliebig viele
+Blockchains von Communities abspeichern kann. Das hängt letzendlich von der Leistung der Hardware ab auf der er läuft und wie groß und aktiv die jeweiligen Communities sind. 
+Damit das System also gut funktioniert braucht es viele Node-Server Betreiber und eventuell müssen wir noch extra Anreize schaffen 
+Gradido-Nodes zu betreiben. 
+Im Idealfall könnte also jeder Gradido Benutzer einen eigenen Node-Server auf seinem Heim-PC betreiben und kann dadurch immer prüfen das seine
+Community nicht manipuliert wurde.
+
+### Wo werden die Keys gespeichert
+In Idealfall würde sich jeder Benutzer um seine eigenen Keys kümmern, so wie es bei Bitcoin und Co gemacht wird. 
+Das ist aber nicht in jedem Fall praktikabel weil die meisten Benutzer nicht so technikaffin sind. 
+Daher das Konzept mit dem Login-Server oder man könnte auch Key-Server dazu sagen, der die Keys möglichst sicher verwahrt.
+So sicher das selbst ein unerfahrender Admin die Sicherheit der Keys nicht so leicht gefährden kann. 
+
+Anfangs sollte er nur die verschlüsselt Keys speichern und dem Benutzer beim Anlegen der Keys eine dazugehörige Passphrase zeigen,
+mit der der User umziehen kann oder sein Passwort zurücksetzen kann, wenn er es mal vergisst. 
+Aber auch das hat sich als zu kompliziert für viele herausgestellt. Daher war die nächste Idee,
+die Passphrase mit dem Admin-Public-Key verschlüsselt zu speichern so das nur der Admin mit seinem Private-Key sie entschlüsseln und damit freigeben kann, für ein Password Reset.
+Der Nachteil das dadurch nun wieder dem Admin vertraut werden muss. Oder man setzte dafür mehrere Vertrauenspersonen ein, welche bei angeforderten Passwort-Reset ihre Keys eingeben. Solange das Programm nicht modifiziert wird, können diese die Private Keys auch nicht missbrauchen. 
+Aktuell wird die Passphrase jedoch noch unverschlüsselt gespeichert. Das verschlüsselte Speichern der Passphrase setzt einige andere Änderungen voraus, und der Passwort-Reset Prozess muss überarbeitet werden. 
```

Comment:
And would require the presence of the admin to change or reset the password. This is not applicable for our software.